h2o-core/src/main/java/hex/grid/GridSearch.java

```diff
@@ -379,7 +379,19 @@ private void gridSearch(Grid<MP> grid) {
               ScoringInfo.sort(grid.getScoringInfos(), sortingMetric());
             }
           } catch (RuntimeException e) { // Catch everything
-            if (!Job.isCancelledException(e)) {
+            if (Job.isCancelledException(e)) {
+              assert model == null;
```

Comment:
is this assertion always true when using CV?
I think so, but I'm not 100% sure: it only depends on step 7 of CV building (`cv_mainModelScores`) never throwing any `CancelledException`.
I think it's ok, just want confirmation.

Reply:
Yes, always true. `buildModel` either returns a model or throws an exception. If it throws an exception `model` will not be defined. The code after `buildModel` should never result in JobCancelledException. The assertion makes sure that stays true in the future as well. Otherwise, I would delete the model in DKV while keeping the model object around. Not the worst thing but better be safe.